wallet-test/src/test/scala/org/bitcoins/wallet/WalletUnitTest.scala

```diff
@@ -186,4 +195,97 @@ class WalletUnitTest extends BitcoinSWalletTest {
+      addrDb <- wallet.addressDAO.findAddress(addr).map(_.get)
+      walletKey = addrDb.ecPublicKey
+      walletPath = addrDb.path
+
+      spk = MultiSignatureScriptPubKey(2, Vector(dummyKey, walletKey))
+      dummyPrevTx = dummyTx(spk = spk)
+      prevTxDb = TransactionDbHelper.fromTransaction(dummyPrevTx)
+      _ <- wallet.transactionDAO.create(prevTxDb)
+
+      psbt = dummyPSBT(prevTxId = dummyPrevTx.txId)
+        .addKeyPathToInput(accountDb.xpub, walletPath, walletKey, 0)
+
+      signed <- wallet.signPSBT(psbt)
+    } yield {
+      assert(signed != psbt)
+      assert(
+        signed.inputMaps.head.partialSignatures.exists(_.pubKey == walletKey))
+    }
+  }
+
+  it must "be able to sign a psbt with our own p2pkh utxo" in {
+    wallet: Wallet =>
+      for {
+        addr <- wallet.getNewAddress(AddressType.Legacy)
+        addrDb <- wallet.addressDAO.findAddress(addr).map(_.get)
+        walletKey = addrDb.ecPublicKey
+
+        spk = addr.scriptPubKey
+        _ = assert(spk == P2PKHScriptPubKey(walletKey))
+        dummyPrevTx = dummyTx(spk = spk)
+        _ <- wallet.processTransaction(dummyPrevTx, blockHashOpt = None)
+
+        psbt = dummyPSBT(prevTxId = dummyPrevTx.txId)
+
+        signed <- wallet.signPSBT(psbt)
+      } yield {
+        assert(signed != psbt)
+        assert(
+          signed.inputMaps.head.partialSignatures.exists(_.pubKey == walletKey))
+      }
+  }
+
+  it must "be able to sign a psbt with our own p2sh segwit utxo" in {
+    wallet: Wallet =>
+      for {
+        addr <- wallet.getNewAddress(AddressType.NestedSegWit)
+        addrDb <- wallet.addressDAO.findAddress(addr).map(_.get)
+        walletKey = addrDb.ecPublicKey
+
+        spk = addr.scriptPubKey
+        _ = assert(spk == P2SHScriptPubKey(P2WPKHWitnessSPKV0(walletKey)))
+        dummyPrevTx = dummyTx(spk = spk)
+        _ <- wallet.processTransaction(dummyPrevTx, blockHashOpt = None)
+
+        psbt = dummyPSBT(prevTxId = dummyPrevTx.txId)
+
+        signed <- wallet.signPSBT(psbt)
+      } yield {
+        assert(signed != psbt)
+        assert(
+          signed.inputMaps.head.partialSignatures.exists(_.pubKey == walletKey))
+      }
+  }
+
+  it must "be able to sign a psbt with our own p2wpkh utxo" in {
+    wallet: Wallet =>
+      for {
+        addr <- wallet.getNewAddress(AddressType.SegWit)
+        addrDb <- wallet.addressDAO.findAddress(addr).map(_.get)
+        walletKey = addrDb.ecPublicKey
+
+        spk = addr.scriptPubKey
+        _ = assert(spk == P2WPKHWitnessSPKV0(walletKey))
+        dummyPrevTx = dummyTx(spk = spk)
+        _ <- wallet.processTransaction(dummyPrevTx, blockHashOpt = None)
+
+        psbt = dummyPSBT(prevTxId = dummyPrevTx.txId)
+          .addUTXOToInput(dummyPrevTx, 0)
+
+        signed <- wallet.signPSBT(psbt)
+      } yield {
+        assert(signed != psbt)
+        assert(
+          signed.inputMaps.head.partialSignatures.exists(_.pubKey == walletKey))
+      }
+  }
```

Comment:
probably should add a test case for signing when we don't have any UTXOs in the PSBT. If I understand correctly, our current implementation would just return the original PSBT right?